mmstudio/src/main/java/org/micromanager/internal/dialogs/PixelPresetEditor.java

```diff
@@ -154,15 +184,28 @@ public boolean writeGroup(String newName) {
          }
 
          core_.definePixelSizeConfig(newName);
-
-         for (PropertyItem item : data_.getProperties()) {
-            if (item.confInclude) {
-               core_.definePixelSizeConfig(newName, item.device, item.name, item.getValueInCoreFormat());              
+         // if no duplicate presets will be created, then allow this preset to be added
+         if (!same) {
+            for (PropertyItem item : data_.getProperties()) {
+               if (item.confInclude) {
+                  core_.definePixelSizeConfig(newName, item.device, item.name, item.getValueInCoreFormat());
+               }
+            }
+            pixelSize_ = pixelSizeField_.getText();
+            core_.setPixelSizeUm(newName, NumberUtils.displayStringToDouble(pixelSize_));
+            core_.setPixelSizeAffine(newName, affineEditorPanel_.getAffineTransform() );
+         }
+         // if fails to edit an existing preset due to the duplicate presets, keep the old preset
+         else if(!newItem_) {
+            for (PropertyItem item : data_.getProperties()) {
+               if (item.confInclude) {
+                  core_.definePixelSizeConfig(newName, item.device, item.name, oldPreset.getSetting(item.device, item.name).getPropertyValue());
+               }
             }
+            pixelSize_ = pixelSizeField_.getText();
+            core_.setPixelSizeUm(newName, NumberUtils.displayStringToDouble(pixelSize_));
+            core_.setPixelSizeAffine(newName, affineEditorPanel_.getAffineTransform() );;
```

Comment:
Note that these lines are nearly identical to the lines above. It is best to avoid repeating code like this, because it makes the code harder to maintain (for example, multiple places will need to be updated when making a change, and it is easy to forget to do so).

In this case, however, I would avoid these lines entirely, instead giving control back to the user: after editing an existing preset A and making its properties identical to preset B, the user should stay in the preset editor after the error message instead of canceling the edit. They can then explicitly cancel, or chose different property values.